crates/manifest/src/schema/v2.rs

```diff
@@ -102,6 +102,9 @@ pub struct Component {
     /// `allowed_http_hosts = ["example.com"]`
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
     pub allowed_http_hosts: Vec<String>,
+    /// `allowed_outbound_hosts = ["myredishost.com"]`
+    #[serde(default, skip_serializing_if = "is_none_or_empty")]
+    pub allowed_outbound_hosts: Option<Vec<String>>,
```

Comment:
It's my understanding that v1 semantics have to be representable in the v2 manifest struct. This is because, we only deserialize into v1. V1 is then immeadiately converted into v2 and the rest of the code only deals with v2 manifests. 

For the host implementation of redis, we need to know whether the `allowed_outbound_hosts` was set or not , because there is a semantic difference between not set at all and set and empty (i.e., if it's set and empty, we deny everything, but if it's not set at all, we allow all). 

cc @lann for any corrections to my understanding.

Reply:
True, though we could translate v1 `None` to `["redis://*:*"]`?

Reply:
We currently don't support '*' globs so this isn't currently possible.